server/schedulers/utils.go

```diff
@@ -458,3 +490,165 @@ func toHotPeerStatShow(p *statistics.HotPeerStat, kind rwType) statistics.HotPee
+					// TiFlash is currently unable to report statistics in the same unit as Region,
+					// so it uses the sum of Regions. If it is not accurate enough, use sum of hot peer.
+					if isTraceRegionFlow {
+						loads[statistics.ByteDim] = storeLoads[statistics.StoreRegionsWriteBytes]
+						loads[statistics.KeyDim] = storeLoads[statistics.StoreRegionsWriteKeys]
+					} else {
+						loads[statistics.ByteDim] = peerLoadSum[statistics.ByteDim]
+						loads[statistics.KeyDim] = peerLoadSum[statistics.KeyDim]
+					}
+				} else {
+					loads[statistics.ByteDim] = storeLoads[statistics.StoreWriteBytes]
+					loads[statistics.KeyDim] = storeLoads[statistics.StoreWriteKeys]
+				}
+			}
+		}
+
+		if isTiFlash {
+			for i := range allTiFlashLoadSum {
+				allTiFlashLoadSum[i] += loads[i]
+			}
+			allTiFlashCount += 1
+			allTiFlashHotPeersCount += len(hotPeers)
+		} else {
+			for i := range allTiKVLoadSum {
+				allTiKVLoadSum[i] += loads[i]
+			}
+			allTiKVCount += 1
+			allTiKVHotPeersCount += len(hotPeers)
+		}
+
+		// Build store load prediction from current load and pending influence.
+		stLoadPred := (&storeLoad{
+			Loads: loads,
+			Count: float64(len(hotPeers)),
+		}).ToLoadPred(rwTy, info.PendingSum)
+
+		// Construct store load info.
+		loadDetail[id] = &storeLoadDetail{
+			Info:     info,
+			LoadPred: stLoadPred,
+			HotPeers: hotPeers,
+		}
+	}
+
+	calcExpect := func(engine string, allLoadSum []float64, allStoreCnt, allHotPeersCnt int) (expect storeLoad) {
+		allStoreCount := float64(allStoreCnt)
+		allHotPeersCount := float64(allHotPeersCnt)
+		expectLoads := make([]float64, len(allLoadSum))
+		for i := range expectLoads {
+			expectLoads[i] = allLoadSum[i] / allStoreCount
+		}
+		expectCount := allHotPeersCount / allStoreCount
+		expect.Loads = expectLoads
+		expect.Count = expectCount
+		// Debug
+		ty := "exp-byte-rate-" + rwTy.String() + "-" + kind.String()
+		hotPeerSummary.WithLabelValues(ty, engine).Set(expectLoads[statistics.ByteDim])
+		ty = "exp-key-rate-" + rwTy.String() + "-" + kind.String()
+		hotPeerSummary.WithLabelValues(ty, engine).Set(expectLoads[statistics.KeyDim])
+		ty = "exp-query-rate-" + rwTy.String() + "-" + kind.String()
+		hotPeerSummary.WithLabelValues(ty, engine).Set(expectLoads[statistics.QueryDim])
+		ty = "exp-count-rate-" + rwTy.String() + "-" + kind.String()
+		hotPeerSummary.WithLabelValues(ty, engine).Set(expectCount)
+		return
+	}
+	tikvExpect := calcExpect(EngineTiKV, allTiKVLoadSum, allTiKVCount, allTiKVHotPeersCount)
+	tiflashExpect := calcExpect(EngineTiFlash, allTiFlashLoadSum, allTiFlashCount, allTiKVHotPeersCount)
+	// store expectation rate and count for each store-load detail.
+	for _, detail := range loadDetail {
+		if detail.Info.IsTiFlash {
+			detail.LoadPred.Expect = tiflashExpect
+		} else {
+			detail.LoadPred.Expect = tikvExpect
+		}
+	}
+
+	return loadDetail
+}
+
+// filterHotPeers filter the peer whose hot degree is less than minHotDegress
+func filterHotPeers(
```

Comment:
What is the relationship between it and `minHotDegree`?

Reply:
Not in the scope of this PR.